File: app/src/main/java/edu/unlpam/vet/ponzonosos/util/Edge.java
package edu.unlpam.vet.ponzonosos.util;

import android.content.Context;
import android.view.View;
import android.view.ViewGroup;

import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;
public class Edge {
    public static void applyDynamicEdgeAppBar(
            Context context,
            View decorView,
            View rootContainer,
            View headerContent,
            float baseHeightDp
    ) {
        ViewCompat.setOnApplyWindowInsetsListener(decorView, (view, insets) -> {
            int statusBarHeight = insets.getInsets(WindowInsetsCompat.Type.statusBars()).top;

            // Setear nueva altura del contenedor
            ViewGroup.LayoutParams layoutParams = rootContainer.getLayoutParams();
            layoutParams.height = (int)(Measures.dpToPx(context, baseHeightDp) + statusBarHeight);
            rootContainer.setLayoutParams(layoutParams);

            // Setear padding superior en el header
            headerContent.setPadding(
                    headerContent.getPaddingLeft(),
                    statusBarHeight,
                    headerContent.getPaddingRight(),
                    headerContent.getPaddingBottom()
            );

            return insets;
        });
    }

}

File: app/src/main/java/edu/unlpam/vet/ponzonosos/util/Edge.kt
package edu.unlpam.vet.ponzonosos.util

import android.content.Context
import android.view.View
import android.view.Window
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat

object Edge {

    @JvmStatic
    fun applyDynamicEdgeAppBar(
        context: Context,
        window: Window,
        rootContainer: View,
        headerContent: View,
        baseHeightDp: Float
    ) {
        ViewCompat.setOnApplyWindowInsetsListener(window.decorView) { _, insets ->
            val statusBarHeight = insets.getInsets(WindowInsetsCompat.Type.statusBars()).top

            // Cambia altura del contenedor
            val layoutParams = rootContainer.layoutParams
            layoutParams.height = (Measures.dpToPx(context, baseHeightDp) + statusBarHeight).toInt()
            rootContainer.layoutParams = layoutParams

            // Cambia padding del header interno
            headerContent.setPadding(
                headerContent.paddingLeft,
                statusBarHeight,
                headerContent.paddingRight,
                headerContent.paddingBottom
            )

            insets
        }
    }
}
